include/openthread/platform/crypto.h

```diff
@@ -170,8 +173,9 @@ typedef struct otPlatCryptoSha256Hash otPlatCryptoSha256Hash;
  */
 typedef struct otPlatCryptoEcdsaKeyPair
 {
-    uint8_t mDerBytes[OT_CRYPTO_ECDSA_MAX_DER_SIZE];
-    uint8_t mDerLength;
+    uint8_t  mDerBytes[OT_CRYPTO_ECDSA_MAX_DER_SIZE]; ///< Used when using literal keys, needs mKeyRef to be 0.
+    uint8_t  mDerLength;                              ///< Length of the stored DER.
+    uint32_t mKeyRef; ///< The PSA key ref (used with OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE).
 } otPlatCryptoEcdsaKeyPair;
```

Comment:
I notcied that we will have a problem with changing `otPlatCryptoEcdsaKeyPair` struct (on platforms that don't use `PLATFORM_KEY_REFERENCES_ENABLE`):

- The SRP ECDA key pair is stored in non-volatile settings (under 
- It is saved as `Crypto::Ecdsa::P256::KeyPair`  which is mapped to `otPlatCryptoEcdsaKeyPair`
- Adding new field `mKeyRef` to this struct will break the backward compatibility when we read the previous value saved in settings.

I think we should keep `otPlatCryptoEcdsaKeyPair` as before. It is always the literal key pair. We can track a key ref in other APIs when we need to deal with it.

We followed the same model for other keys, e.g. `otNetworkKey` is always the literal key and we then defined a separate type `otNetworkKeyRef` when representing it as key ref in PSA.

Reply:
Ok, looks like I will have to re-think these changes again. I will close this PR and open a new one in its place.

Reply:
Sounds good. Thanks. Sorry! I did not realize this earlier.

Reply:
No issues at all. Sounds like a potential problem and we need to address this.